controllers/admin/AdminSelfUpgradeController.php

```diff
@@ -151,17 +151,20 @@ public function __construct()
                 return;
             }
 
-            // If a previous version of ajax-upgradetab.php exists, delete it
-            if (file_exists($this->autoupgradePath . DIRECTORY_SEPARATOR . 'ajax-upgradetab.php')) {
-                @unlink($this->autoupgradePath . DIRECTORY_SEPARATOR . 'ajax-upgradetab.php');
-            }
-
+            clearstatcache();
```

Comment:
Is this call necessary if the file is not altered or checked before this line?

Reply:
After reading PHP manual for clearstatcache, I added it as a precaution.
But, as you say, the file is not altered or checked before this line, so it sems to be not required.

Also, I tested on my server by commenting out line 154, and there was no problem.

So I would say that it is not necessary to call `clearstatcahe()` at that point.

Reply:
Given that it is not mandatory to fix your issue, if you don't mind removing it from the PR I will approve the rest of the changes.

Reply:
Done